utils/log.go

```diff
@@ -0,0 +1,96 @@
+package utils
+
+import "fmt"
+
+func FormatWritesLog(codes []byte, line *string) {
+	n := len(codes)
+	str := ""
+	exist := false
+	if n == 3 {
```

Comment:
This will work most of the time, but if the client is pasting a bunch of input rather than typing it one character at a time, you'll miss out on the nice formatting here.

Reply:
Also please add a comment explaining what `n == 3` signifies (i.e. arrow keys).